### PR TITLE
Drives list and Rides list

### PR DESCRIPTION
**Pull request by istzen**
add new Model trip. Both Ride and Drive extend trip. 
add generic list view and implement for Rides and Drives
add cards for drives, rides but their concrete design is part of another PR
move create drive form back into create_drive_page.

### CODE REVIEW

**Review by Showcas — APPROVED** [review]

**Comment by fynsta** [comment]
I'm wondering if the dates should be part of the cards, or if there should be a labeled separator for each day

**Comment by fynsta** [comment]
I definitely like the nice generalization 👍

**Comment by Eddie-42** [comment]
> I definitely like the nice generalization 👍

I second that!

**Comment by istzen** [comment]
> I'm wondering if the dates should be part of the cards, or if there should be a labeled separator for each day
> 

I thought about that too, but then again especially for the rides, i don't think their will be more than one or two cards per user, so i think having the date with the cards is not a problem. Also it might look weird when there is one Card an a Date above it.

**Comment by istzen** [comment]
> 
> I think I would prefer this, too. Also, the cards are really not that beautiful right now and we need to integrate the pre-loading functionality of the DriveDetailPage so I think that's a new user story right there :D For now, this is fine right?

Yes i like the idea of a new Userstory, i think for now the cards are fine, if we think of them as placeholders.

**Review by fynsta — CHANGES_REQUESTED** [review]

File: lib/util/trip/drive_card.dart [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                    padding: const EdgeInsets.all(32.0),
+                    child: Row(
+                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                      children: [
+                        Text('${formatTime(trip.endTime)}  ${trip.end}'),
+                      ],
+                    ),
+                  ),
+                  node: const TimelineNode(
+                    indicator: OutlinedDotIndicator(),
```
[the remark this answers is not in the document]
Comment (istzen) [reply lib/util/trip/drive_card.dart:60]: yes I will do that

File: lib/util/trip/trip_page_builder.dart [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            ],
+          ),
+        ),
+        body: TabBarView(
+          children: [
+            TripStreamBuilder<T>(
+              stream: trips,
+              emptyMessage: 'No upcoming $name',
+              filterTrips: (trips) => trips
+                  //we could change this to trip.starttime the question is what we are doing with
```
[the remark this answers is not in the document]
Comment (istzen) [reply lib/util/trip/trip_page_builder.dart:38]: yes i think this comment can be deleted

File: lib/drives/pages/drives_page.dart [hunk 3]
```diff
@@ -9,16 +14,29 @@ class DrivesPage extends StatefulWidget {
 }
 
 class _DrivesPageState extends State<DrivesPage> {
+  late final Stream<List<Drive>> _drives;
+
+  @override
+  void initState() {
+    //todo: method to get userId
```
[the remark this answers is not in the document]
Comment (istzen) [reply lib/drives/pages/drives_page.dart:20]: yes, thanks for the reminder

File: lib/util/trip/trip_page_builder.dart [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                text: 'Past',
+              ),
+            ],
+          ),
+        ),
+        body: TabBarView(
+          children: [
+            TripStreamBuilder<T>(
+              stream: trips,
+              emptyMessage: 'No upcoming $name',
```
[the remark this answers is not in the document]
Comment (istzen) [reply lib/util/trip/trip_page_builder.dart:36]: so what do you suggest?

[the remark this answers is not in the document]
Comment (fynsta) [reply lib/util/trip/trip_page_builder.dart:32]: I don't see a better way either, so we can leave it like this... the person who adds localization will have to think about this lol

**Comment by fynsta** [comment]
you wanted to revert the ios files, not the macos ones 🤣

**Comment by istzen** [comment]
> you wanted to revert the ios files, not the macos ones 🤣

yes, i did revert the ios files the descriotion of the commit is just wrong i'm sorry

**Review by fynsta — APPROVED** [review]

**Pull request merged**